### ISSUE TITLE
Update instructions for testing in QEMU

### ISSUE THREAD

**Issue by GBA-17** [u-root/webboot#189]
**Is your feature request related to a problem? Please describe.**
The README instructions for testing in QEMU are currently out of date. We need to figure out the arguments for QEMU to locate the USB stick so that we can use the ISO cache.

**Describe the solution you'd like**
When we start up `webboot` in QEMU, we should be able to locate the USB stick containing the ISO cache.

**Describe alternatives you've considered**
One configuration we tried was:
```sh
qemu-system-x86_64 -m 2G -hda /dev/sda
```

This allowed us to boot into the kernel and initrd on the USB stick, but we couldn't access the USB stick once we booted into the u-root environment.

**Additional context**
N/A

**Comment by bjakobson** [comment]
Looks like the readme been updated. Can we close the issue?

**Comment by GBA-17** [comment]
I don't think it's ready to close yet. We made a switch this summer from copying the ISO to `pmem0` to accessing the ISO directly from a USB stick. The issue right now is that when we boot into U-root from QEMU, the USB stick doesn't show up in /dev. That keeps us from accessing the ISOs on the USB stick, and from testing Webboot.

**Comment by bjakobson** [comment]
When I was using syslinux, i found that my usb stuck was under /dev/sdb1 instead of /dev/sda. /sda was my VM.

**Comment by GBA-17** [comment]
Do you mean that there were multiple USB sticks plugged in? Like one USB stick had syslinux and the u-root stuff, and the other had files you wanted to access?

**Comment by bjakobson** [comment]
Nope, just had my VM and my USB stick. Then using QEMU, I had to use /sdb1. Can you confirm that your current USB stick partition is /sda and not /sdb?

**Comment by GBA-17** [comment]
On my main Linux system, my USB stick is at /dev/sdb. What parameters are you using when you boot up QEMU?

**Comment by bjakobson** [comment]
It's been a few months since I've tried it. I can find my commands

**Comment by bjakobson** [comment]
I used @orangecms shell program to launch QEMU. run-webboot.sh

**Comment by orangecms** [comment]
With the `-hda`, did you run a kernel that supports the necessary storage drivers? I can think of AHCI. It is different from the USB mass storage driver. And you will need to support the filesystem as well.

See also https://wiki.gentoo.org/wiki/QEMU/Options#Hard_drive

**Comment by bjakobson** [comment]
I don't quite know. I ran your shell script, then selected "b" from the graphics option and then i was able to use webboot. 


Bare in mind, this was probably 3 months ago 😀

**Comment by GBA-17** [comment]
Ok, good points about the shell script and AHCI.

I think I got it working with this command. Once QEMU started, I located the USB stick in u-root, and was able to use Webboot to boot into Tinycore.

```sh
qemu-system-x86_64 -m 2G -drive id=disk,file=/dev/sdb,if=none -device ahci,id=ahci -device ide-drive,drive=disk,bus=ahci.0
```
Except there's a few issues:
- I copied and pasted from the link @orangecms shared, so I'm not totally sure what it all means
- There's no network access. I think some of the parameters from the shell script would fix that, but I don't completely understand all the arguments there

**Comment by bjakobson** [comment]
So /dev/sdb worked for you? 

@orangecms will be able to give you a far better explanation about his shell script, but here is probably the most important line for wifi configuration:

'netdev user,id=network0 -device rtl8139,netdev=network0'
